build-env/inc/env-common.sh

```diff
@@ -1,24 +1,24 @@
-OVERRIDE_BIN_PATH=/opt/bin
+OVERRIDE_PYTHON_BIN_PATH=/opt/bin
 
-export PATH=${OVERRIDE_BIN_PATH}:${PATH}:/usr/lib/pe-golang/bin:/usr/lib/pelion/bin
+export PATH=${OVERRIDE_PYTHON_BIN_PATH}:${PATH}:/usr/lib/pe-golang/bin:/usr/lib/pelion/bin
 
-# link python to python2 or python3 in $OVERRIDE_BIN_PATH dir
+# link python to python2 or python3 in $OVERRIDE_PYTHON_BIN_PATH dir
 # arg: 2 or 3
 function select_python()
 {
     local PY=$(which python"$1")
 
-    if [ $? -ne 0 ];then
+    if [ ! -n "$PY" ];then
         echo "ERROR: cannot find python"$1" in PATH"
         return 1
     fi
 
-    if ! stat $OVERRIDE_BIN_PATH >/dev/null 2>&1; then
-        sudo mkdir -p $OVERRIDE_BIN_PATH
+    if ! stat $OVERRIDE_PYTHON_BIN_PATH >/dev/null 2>&1; then
+        sudo mkdir -p $OVERRIDE_PYTHON_BIN_PATH
     fi
 
     echo "Selecting python"$1" as default python interpreter"
-    sudo ln -fs $PY $OVERRIDE_BIN_PATH/python
+    sudo ln -fs $PY $OVERRIDE_PYTHON_BIN_PATH/python
 }
```

Comment:
This symlink stuff seems convoluted. For anything requiring python2, shouldn't we simply change it to run `python2` instead of `python`?  That seems simpler, more explicit, and more standard.

Reply:
The script which is failing is libuv dependency inside greasgo repo. It would be best to fix it in there - python2 is deprecated for more than a year now, so best option would be update scripts so it can be run on supported python version. 

But it looks like it 'python' link is back in `focal` packages - so setting it explicitly will not be needed. @sameer2209-arm please confirm if it compiles on dev branch without this change.

Reply:
@kbgl maestro build is still failing for focal on the dev branch

Reply:
@ryannowarm there are options:
1. we can try to create a patch for maestro sources in distro-pelion-edge so all `*.py` files will use `python2`
2. we can keep it as it is except splitting the workaround from other changes - there is one fix inside `select_python` function which we should keep anyway; do fix in `greasego` repository (use `python2` or upgrade to `python3`) and then revert workaround
3. just keep as it is now

What do you prefer?

Reply:
> * we can try to create a patch for maestro sources in distro-pelion-edge so all `*.py` files will use `python2`
I like this idea the best.  Perhaps we can even submit a change to the maestro repo itself so we don't have to maintain a distro-specific patch?


> * we can keep it as it is except splitting the workaround from other changes - there is one fix inside `select_python` function which we should keep anyway; do fix in `greasego` repository (use `python2` or upgrade to `python3`) and then revert workaround

This seems an ok option as well.

> * just keep as it is now

I'm fine with this option too as long as things work